### PR TITLE
add gcp provider and generate addon

### PR DESCRIPTION
**Pull request by Nicola115**
Signed-off-by: Avery Qi <2225992901@qq.com>
### Description of your changes
add provider-GCP.yaml and generate gcp addon.

### Checklist
I have:
- [ ] Title of the PR starts with OAM type (e.g. `[Workload]` or `[Trait]` or `[Scope]`).
- [ ] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

### CODE REVIEW

**Comment by msftgits** [comment]
[CLA assistant check](https://cla.opensource.microsoft.com/oam-dev/catalog?pullRequest=245) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/oam-dev/catalog?pullRequest=245) before we can accept your contribution.<br/><br/>:x: Nicola115 [sign now](https://cla.opensource.microsoft.com/oam-dev/catalog?pullRequest=245)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/oam-dev/catalog?pullRequest=245) it.</sub>

File: hack/addons/terraform/provider-GCP.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# The Cloud name of the provider
+cloudName: Google Cloud Platform
+
+# When enabling a Terraform provider, these properties need to set for authentication. For Tencent Cloud,
+# name: Environment variable name when authenticating Terraform, like https://github.com/oam-dev/terraform-controller/blob/master/controllers/provider/credentials.go#L59
+# secretKey: Secret key when storing authentication information in a Kubernetes, like https://github.com/oam-dev/terraform-controller/blob/master/controllers/provider/credentials.go#L109.
+properties:
+  - name: GOOGLE_CREDENTIALS
+    secretKey: gcpCredentialsJSON
+    description: Get gcpCredentialsJSON per this guide https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/getting_started#adding-credentials
```
Comment (zzxwill) [inline hack/addons/terraform/provider-GCP.yaml:13]:
```suggestion
    description: Get GOOGLE_CREDENTIALS per this guide https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/getting_started#adding-credentials
```

File: hack/addons/terraform/provider-GCP.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - name: GOOGLE_REGION
+    description: Get GOOGLE_REGION by picking one RegionId from Google Cloud region list https://cloud.google.com/compute/docs/regions-zones
+    isRegion: true
+  - name: GOOGLE_PROJECT
+    secretKey: gcpProject
+    description: Set gcpProject per this guide https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/getting_started#configuring-the-provider
```
Comment (zzxwill) [inline hack/addons/terraform/provider-GCP.yaml:19]:
```suggestion
    description: Set GOOGLE_PROJECT per this guide https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/getting_started#configuring-the-provider
```

**Review by zzxwill — APPROVED** [review]
Great job!

**Review by wonderflow — APPROVED** [review]
thanks

**Comment by zzxwill** [comment]
@barnettZQG Can you take a look at this issue?

**Comment by barnettZQG** [comment]
@iamrajiv 
```
vela port-forward -n vela-system addon-velaux 9082:80
```

**Pull request merged**